lib/watcher_adapter.js

```diff
@@ -27,43 +27,40 @@ class WatcherAdapter extends EventEmitter {
     if (!Array.isArray(watchedPaths)) {
       throw new TypeError(`WatcherAdapter#watch's first argument must be an array of watchedPaths`);
     }
-
     let watchers = watchedPaths.map(watchedPath => {
-      return new Promise(resolve => {
-        const watcher = new sane(watchedPath, this.options);
-        this.watchers.push(watcher);
-
-        bindFileEvent(this, watcher, 'change');
-        bindFileEvent(this, watcher, 'add');
-        bindFileEvent(this, watcher, 'delete');
-
+      const watcher = new sane(watchedPath, this.options);
+      this.watchers.push(watcher);
+      bindFileEvent(this, watcher, 'change');
+      bindFileEvent(this, watcher, 'add');
+      bindFileEvent(this, watcher, 'delete');
+      return new Promise((resolve, reject) => {
+        watcher.on('ready', resolve);
+        watcher.on('error', reject);
+      }).then(() => {
+        watcher.removeAllListeners('ready');
+        watcher.removeAllListeners('error');
         watcher.on('error', err => {
           logger.debug('error', err);
           this.emit('error', err);
         });
-
-        watcher.on('ready', () => {
-          logger.debug('ready', watchedPath);
-          resolve(watcher);
-        });
+        logger.debug('ready', watchedPath);
       });
     });
-
-    return Promise.all(watchers).then(function() {});
+    return Promise.all(watchers).then(() => {});
   }
 
   quit() {
-    let closing = this.watchers.map(watcher => {
-      return new Promise((resolve, reject) => {
-        watcher.close(err => {
-          if (err) reject(err);
-          else resolve();
-        });
-      });
-    });
+    let closing = this.watchers.map(
+      watcher =>
+        new Promise((resolve, reject) =>
+          watcher.close(err => {
+            if (err) reject(err);
+            else resolve();
+          })
+        )
+    );
     this.watchers.length = 0;
-
-    return Promise.all(closing).then(function() {});
+    return Promise.all(closing).then(() => {});
```

Comment:
Just curious, why do we need the empty function here?

Reply:
There is a test for `Promise<undefined>` vs `Promise<undefined[]>` not sure why it wasn't failing before.  I believe there are still issues with the promises, I would like to convert to typescript so we can use the tslint no-floating-promises